src/Catalog/Telemetry/ITelemetryService.cs

```diff
@@ -11,6 +11,8 @@ namespace NuGet.Services.Metadata.Catalog
 {
     public interface ITelemetryService
     {
+        TelemetryClientWrapper TelemetryClient { get; }
```

Comment:
why is this needed? Isn't the goal of this interface to "hide" the implementation details of telemetry logging?

Reply:
The `LightningJob` is the only one requiring a singleton `TelemetryClient` to be [registered in the ServicesCollection](https://github.com/NuGet/NuGet.Services.Metadata/pull/732/files#diff-7b3387943c5f27215edf918eac4da605R525), but in order to do so, I need to be able to access the singleton created during ApplicationInsights initialization.

It's either this change, or I need to pass in the `TelemetryClient` to the constructor of all jobs in addition to the `ITelemetryService`.

Reply:
Actually, I'm wondering why this particular job needs this `TelemetryClient` to be available in the `ServicesCollection`... No idea which type is consuming this as a constructor argument.

@joelverhagen you any idea why this might be the case? It got introduced as part of [this commit](https://github.com/NuGet/NuGet.Services.Metadata/commit/68f7b9a12c737076689857dfb8f135ef99d96c85). My gut feeling says it has to do with the `IV3TelemetryService` that is possibly used by this job which requires another `TelemetryClient`, whereas we manually construct the other `ITelemetryService` and pass it to the job constructors of all jobs...

Reply:
Did a refactoring of how `TelemetryService` is used and constructed in commit https://github.com/NuGet/NuGet.Services.Metadata/pull/732/commits/09a0ae7e27cac32c91e249017d4acf52deaca395.